### PR TITLE
[ADD,FIX] l10n_xi, l10n_uk: Northern Ireland accounting module, update UK taxes and report for brexit 

### PR DESCRIPTION
**Pull request by dbkosky**
To clarify here:
The UK consists of England, Northern Ireland, Scotland and Wales
Great Britain consists of: England, Scotland and Wales

The current arrangement is that Northern Ireland will trade with the European Union member states in goods as though it were a European Union member state itself (following EU VAT rules), and trade services as though it's not (following domestic VAT rules). Northern Ireland will trade goods and services with the Great Britain following domestic VAT rules.

In base, Northern Ireland is added as a country, with the country code XI, it's subsequently added to the the Europe country group in base (to clarify, this is to make European transactions easier, it is not a political statement).

The l10n_xi module has been created to cover the specifics above for Northern Ireland, adding new taxes and fiscal positions for trade with Europe. A new chart of accounts for Northern Ireland inherits from l10n_uk, so it should share all the UK taxes, fiscal positions, tax report, etc.

In l10n_uk, the titles of the tax report lines have also been updated to match what is expected in the HMRC tax report. Taxes had to be modified and taxes had to be added in the UK chart of accounts in order to differentiate between taxes on goods and taxes on services. New taxes had to be added to reflect UK trade with the "rest of the world", which should apply to all UK exports and imports. Fiscal positions for the UK chart of accounts were added to reflect domestic trade and UK trade with the rest of the world.

task-2675406

### CODE REVIEW

File: addons/l10n_uk/data/account.tax.group.csv [hunk 1]
```diff
@@ -1,5 +1,10 @@
 id,name,country_id/id
 tax_group_0,TAX 0%,base.uk
 tax_group_5,TAX 5%,base.uk
+tax_group_125,TAX 12.5%,base.uk
 tax_group_175,TAX 17.5%,base.uk
 tax_group_20,TAX 20%,base.uk
+tax_group_0RC,Reverse Charge: Customer to pay VAT to HMRC (0%),base.uk
```
Comment (tsb-odoo) [inline addons/l10n_uk/data/account.tax.group.csv:7]: I don't think we need additional tax groups. One tax group per tax rate is enough.

Comment (dbkosky) [reply addons/l10n_uk/data/account.tax.group.csv:7]: My concern for this is that fact that it's required when you're self billing that you denote the reverse charge on the invoice somewhere. 
See section (10.7) 
https://www.gov.uk/guidance/the-vat-domestic-reverse-charge-procedure-notice-735#para9

**Comment by qdp-odoo** [comment]
@Laura-labe Soooo, there's another PR to dust off here :) Seems like it's for you too ^^

**Comment by dbkosky** [comment]
@qdp-odoo Some of the content of this PR has already been merged (in saas-16.2 onwards). 
Looking back on it, a lot of this PR is bloat and I think this later PR https://github.com/odoo/odoo/pull/132298 does a better job. 
Do you think it's ok to close this PR with reference to https://github.com/odoo/odoo/pull/132298?

**Comment by qdp-odoo** [comment]
@dbkosky hey, you know better :) you're the author of both the PR ^^

**Comment by dbkosky** [comment]
This PR has been superseded by another, slightly smaller PR: https://github.com/odoo/odoo/pull/132298

**Pull request closed without merging**